lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -2508,6 +2516,116 @@ signTransaction tl preferredLatestEra keyLookup (rootKey, rootPwd) utxo =
             keyLookup
             inputResolver
 
+type MakeRewardAccountBuilder k =
+    (k 'RootK XPrv, Passphrase "encryption") -> (XPrv, Passphrase "encryption")
+
+-- | Produce witnesses and construct a transaction from a given selection.
+--
+-- Requires the encryption passphrase in order to decrypt the root private key.
+-- Note that this doesn't broadcast the transaction to the network. In order to
+-- do so, use 'submitTx'.
+--
+buildAndSignTransactionNoSelection
+    :: forall (n :: NetworkDiscriminant)
+     . (Typeable n, DelegationAddress n ShelleyKey 'CredFromKeyK)
+    => Tracer IO WalletLog
+    -> TimeInterpreter (Either PastHorizonException)
+    -> DBLayer IO (SeqState n ShelleyKey) ShelleyKey
+    -> NetworkLayer IO Block
+    -> TransactionLayer ShelleyKey 'CredFromKeyK SealedTx
+    -> WalletId
+    -> AnyRecentEra
+    -> Passphrase "user"
+    -> TransactionCtx
+    -> IO (Tx, TxMeta, SealedTx)
+buildAndSignTransactionNoSelection
+    tr ti db@DBLayer{..} netLayer txLayer
+    walletId era pwd txContext = WriteTx.withRecentEra era $
+    \(_recentEra :: WriteTx.RecentEra recentEra) -> do
+        utxo@(_utxoIndex, wallet, _history) <- atomically $ do
+            checkpoint <- readCheckpoint walletId >>=
+                maybe (throwIO
+                    $ ExceptionConstructTx
+                    $ ErrConstructTxNoSuchWallet
+                    $ ErrNoSuchWallet walletId) pure
+            pending  <-
+                readTxHistory
+                    walletId Nothing Descending wholeRange (Just Pending)
+            let pendingTxs = Set.fromList (fromTransactionInfo <$> pending)
+            pure
+                ( UTxOIndex.fromMap
+                    $ CS.toInternalUTxOMap
+                    $ availableUTxO pendingTxs checkpoint
+                , checkpoint
+                , pendingTxs
+                )
+
+        protocolParams <- currentProtocolParameters netLayer
+        unsignedTxBody <-
+            either (throwIO . (ExceptionConstructTx . ErrConstructTxBody)) pure $
+                mkUnsignedTransaction txLayer @recentEra
+                    (getRawKey (Seq.rewardAccountKey (getState wallet)))
+                    protocolParams txContext (Left (PreSelection []))
```

Comment:
👆 Sure we don't need pass in a `PreSelection`with outputs from `quitStakePool`, but I don't think it should be any more trouble to do add as argument to `buildAndSignTransactionNoSelection` already now